src/models/ComponentStyle.js

```diff
@@ -4,6 +4,33 @@ import hashStr from '../vendor/glamor/hash'
 import type { RuleSet, NameGenerator, Flattener, Stringifier } from '../types'
 import StyleSheet from './StyleSheet'
 
+import isStyledComponent from '../utils/isStyledComponent'
+
+
+const isStaticRules = (rules: RuleSet): boolean => {
+  for (let i = 0; i < rules.length; i += 1) {
+    const rule = rules[i]
+
+    // recursive case
+    if (Array.isArray(rule)) {
```

Comment:
sure! Makes sense that if V8 knows all the branches ahead of time it can better about optimizing.